stac_geoparquet/arrow/_api.py

```diff
@@ -55,14 +105,108 @@ def parse_stac_items_to_arrow(
         )
         return pa.RecordBatchReader.from_batches(schema, batches)
 
-    else:
-        # If schema is _not_ provided, then we must convert to Arrow all at once, or
-        # else it would be possible for a STAC item late in the collection (after the
-        # first chunk) to have a different schema and not match the schema inferred for
-        # the first chunk.
+    elif schema == "FullFile":
         batch = stac_items_to_arrow(items)
+        logger.debug(batch.schema, batch)
         return pa.RecordBatchReader.from_batches(batch.schema, [batch])
 
+    elif schema == "FirstBatch":
+        batches = (
+            stac_items_to_arrow(batch) for batch in batched_iter(items, chunk_size)
+        )
+        return from_batches(batches)
+
+    else:
+        if tmpdir is None or Path(tmpdir).exists() is False:
```

Comment:
This makes it clearer to the reader that we've completed the enum check

```suggestion
        assert schema == "ChunksToDisk"
        if tmpdir is None or Path(tmpdir).exists() is False:
```